### PR TITLE
added new function create_docker

### PR DESCRIPTION
**Pull request by moezali1**
Added a new function `create_docker` that generates `requirements.txt` and `Dockerfile` file.

```
from pycaret.datasets import get_data
data = get_data('iris')

from pycaret.classification import *
s = setup(data, target = 'species', session_id = 123)

lr = create_model('lr')

create_api(lr, 'my_first_api')

create_docker('my_first_api')
```

image

### CODE REVIEW

File: pycaret/classification.py [hunk 1]
```diff
@@ -2997,4 +3018,18 @@ def create_api(estimator, api_name: str, host: str = '127.0.0.1', port: int = 80
...
+        estimator=estimator, api_name=api_name, host=host, port=port
+    )
+
+
+def create_docker(
+    api_name: str, base_image: str = "python:3.8-slim", expose_port: int = 8000
+) -> None:
+
+    """
+    creates docker image
```
Comment (Yard1) [inline pycaret/classification.py:3031]: Needs better docstring

File: pycaret/internal/tabular.py [hunk 2]
```diff
@@ -10538,6 +10538,49 @@ def predict({INPUT_COLS}):
...
+""".format(BASE_IMAGE=base_image, PORT=expose_port, API_NAME=api_name)
+    
+    f = open("Dockerfile", "w")
+    f.write(docker)
+    f.close
+    
+    print("""Dockerfile and requirements.txt successfully created.
+To build image you have to run --> !docker image build -f "Dockerfile" -t IMAGE_NAME:IMAGE_TAG .
+        """)
+
```
Comment (Yard1) [inline pycaret/internal/tabular.py:10583]: return the name of the created file

File: pycaret/internal/tabular.py [hunk 3]
```diff
@@ -10538,6 +10538,49 @@ def predict({INPUT_COLS}):
...
+RUN pip install -r requirements.txt
+
+EXPOSE {PORT}
+
+CMD ["python", "{API_NAME}.py"]    
+""".format(BASE_IMAGE=base_image, PORT=expose_port, API_NAME=api_name)
+    
+    f = open("Dockerfile", "w")
+    f.write(docker)
+    f.close
```
Comment (Yard1) [inline pycaret/internal/tabular.py:10578]:
```suggestion
    with open("Dockerfile", "w") as f:
        f.write(docker)
```

File: pycaret/regression.py [hunk 4]
```diff
@@ -2697,4 +2722,18 @@ def create_api(estimator, api_name: str, host: str = '127.0.0.1', port: int = 80
...
+        estimator=estimator, api_name=api_name, host=host, port=port
+    )
+
+
+def create_docker(
+    api_name: str, base_image: str = "python:3.8-slim", expose_port: int = 8000
+) -> None:
+
+    """
+    creates docker image
```
Comment (Yard1) [inline pycaret/regression.py:2735]: Needs better docstring

**Review by Yard1 — CHANGES_REQUESTED** [review]

**Review by moezali1 — COMMENTED** [review]
added docstrings

**Pull request merged**